samples/bluetooth/mesh_provisioner/src/main.c

```diff
@@ -5,11 +5,15 @@
  */
 
 #include <zephyr/sys/printk.h>
-
 #include <zephyr/settings/settings.h>
-
 #include <zephyr/bluetooth/bluetooth.h>
 #include <zephyr/bluetooth/mesh.h>
+#include <zephyr/drivers/gpio.h>
+
+#define SW0_NODE	DT_ALIAS(sw0)
+#if !DT_NODE_HAS_STATUS(SW0_NODE, okay)
+#error "Unsupported board: sw0 devicetree alias is not defined"
```

Comment:
What about platforms `**qemu_cortex_m3 qemu_x86**`? They are in the allowed platforms list.
Is it still possible to build and run app for them?

Reply:
I'm not sure about the importance of platforms. This is a question mostly to the code owner.
However, I think if it is not possible to build an application for the board it shouldn't be in the list of allowed platforms.

Reply:
(not a code owner of this sample) No, it won't build. But a user may want to run this sample in QEMU or as native application and use BlueZ as a controller. This variant is also mentioned in the documentation. As a suggestion, it can either be compiled with button option for boards that has a button and without button option for other boards (like QEMU).

Reply:
I've done as @PavelVPV suggested and in the latest commit the feature is only enabled if the board supports GPIO. The sample builds for all specified platforms.